portal/src/main/webapp/js/src/DataProxyFactory.js

```diff
@@ -91,8 +91,62 @@ var DataProxyFactory = (function()
 
 		return _defaultMutationDataProxy;
 	}
+	
+	function getCustomMutationDataProxy(geneticProfiles,caseSetId,caseIdsKey,caseList)
```

Comment:
Instead of duplicating the function `getDefaultMutationDataProxy` we should add optional parameters to `getDefaultMutationDataProxy` and modify the function as needed. Duplicated code is not easy to review/maintain.